### PR TITLE
Bump Java Version to 20

### PR DESCRIPTION
**Pull request by java-coding-prodigy**
Bump Java Version of the bot.
Please reflect the same on the website too

### CODE REVIEW

**Comment by java-coding-prodigy** [comment]
Also had to upgrade gradle to `8.3` so there are a few changes from the upgrade as well

**Review by Taz03 — APPROVED** [review]

File: application/src/main/java/org/togetherjava/tjbot/features/mathcommands/wolframalpha/WolframAlphaImages.java [hunk 1]
```diff
@@ -66,10 +65,10 @@ static BufferedImage renderSubPod(SubPod subPod) {
...
         try {
-            graphics.drawImage(ImageIO.read(new URL(sourceImage.getSource())), IMAGE_MARGIN_PX,
-                    IMAGE_MARGIN_PX, null);
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
+            graphics.drawImage(ImageIO.read(new URI(sourceImage.getSource()).toURL()),
+                    IMAGE_MARGIN_PX, IMAGE_MARGIN_PX, null);
+        } catch (IOException | URISyntaxException e) {
+            throw new RuntimeException(e);
         }
```
Comment (Taz03) [inline application/src/main/java/org/togetherjava/tjbot/features/mathcommands/wolframalpha/WolframAlphaImages.java:72]: ig u'll have to catch and throw both separately

**Review by Taz03 — CHANGES_REQUESTED** [review]

File: application/src/main/java/org/togetherjava/tjbot/features/mathcommands/wolframalpha/WolframAlphaImages.java (hunk 1 above)
Comment (Zabuzard) [reply application/src/main/java/org/togetherjava/tjbot/features/mathcommands/wolframalpha/WolframAlphaImages.java:72]: why are these code changes in this PR? are they required for java 20?

**Review by Zabuzard — CHANGES_REQUESTED** [review]

**Comment by java-coding-prodigy** [comment]
The changes in `WolframAlphaImages.java` are required because `URL#<init>(String` is deprecated in Java 20 and the suggested alternative is to use `URI#toURL`.
This also changes the exception handling a little bit but functionally it is the same as what we were doing previously.

**Comment by Zabuzard** [comment]
make spotless and codeql happy please

**Comment by java-coding-prodigy** [comment]
> make spotless and codeql happy please

For CodeQL to be fixed we need #903 
Spotless is also having some weird error while trying to build, I’ll have to check that.

**Comment by java-coding-prodigy** [comment]
> > make spotless and codeql happy please
> 
> For CodeQL to be fixed we need #903 Spotless is also having some weird error while trying to build, I’ll have to check that.

Apparently that isn’t the issue.
For some reason Gradle 8.3 is not able to run files compiled by Java 20 even though it is supposed to support Java 20
image

**Review by Taz03 — APPROVED** [review]

**Review by Zabuzard — APPROVED** [review]

**Pull request merged**